### ISSUE TITLE
Function with slash "/" in the name is not working on Azure

### ISSUE THREAD

**Issue by Ungerfall** [Azure/azure-functions-dotnet-worker#777]
I have multiple HTTP triggered with Route functions in a single C# class:
``` csharp
...
[Function("GET /api/modules")]
public async Task<HttpResponseData> GetModules([HttpTrigger(AuthorizationLevel.Function, "get", Route = "modules")] HttpRequestData req)
{
...
}

[Function("POST /api/tasks/{id}/solution")]
public async Task<HttpResponseData> CheckSolution(
        [HttpTrigger(AuthorizationLevel.Function, "post", Route = "tasks/{id}/solution")] HttpRequestData req,
        string id)
{
...
}
```

Locally works fine, but on Azure I'm getting 404 error.

Changing a name fixes the problem. For example, from `"GET /api/modules"` to `"GET modules"`

**Comment by kshyju** [comment]
Hi @Ungerfall What request URL were you getting 404 for? The code you shared above looks fine. I tried your code both locally and azure (deployed to windows consumption plan) and it works fine (I did not get 404 for the GET and POST endpoint). Also, what change fixed your problem?

**Comment by Ungerfall** [comment]
Hello @kshyju I just checked with `AuthorizationLevel.Anonymous` and it works. However, using `AuthorizationLevel.Function` is unable to retrieve the function key. In the Azure portal it is constantly loading and `func azure functionapp publish` or `func azure functionapp list-functions --show-keys` returns URL without code so it gets 401 eventually

**Comment by kshyju** [comment]
You can get the keys from azure portal and use that when invoking the function. See various ways to get keys here: [Obtaining keys
](https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-http-webhook-trigger?tabs=csharp#obtaining-keys)

Could you point me to the documentation you are following, which suggests you that you will get the keys with the 2 approaches you tried. There may be a documentation gap we may need to fix. Thanks!

**Comment by kshyju** [comment]
@Ungerfall The only special characters allowed in the function name value is `-` and `_`. We do have a validation method which validates it against [the defined format](https://github.com/Azure/azure-functions-host/blob/dcfd7fc1458da020de2c3f49b1d9672b1e1bc229/src/WebJobs.Script/Utility.cs#L46). But looks like it is not being called for this out of process worker code path. We will fix that in a future version.

To mitigate your issue, please use a function name which adheres to the format defined in [this regex expression](https://github.com/Azure/azure-functions-host/blob/dcfd7fc1458da020de2c3f49b1d9672b1e1bc229/src/WebJobs.Script/Utility.cs#L46). Thanks!

**Comment by kshyju** [comment]
We will track the work needed to fix the issue in #788
